sklearn/utils/_estimator_html_repr.py

```diff
@@ -78,24 +80,81 @@ def _write_label_html(
+    name : str
+        The label for the estimator. It corresponds either to the estimator class name
+        for simple estimator or in the case of `Pipeline` and `ColumnTransformer`, it
+        corresponds to the name of the step.
+    name_details : str
+        The details to show as content in the dropdown part of the toggleable label.
+        It can contain information as non-default parameters or column information for
+        `ColumnTransformer`.
+    outer_class : {"sk-label-container", "sk-item"}, default="sk-label-container"
+        The CSS class for the outer container.
+    inner_class : {"sk-label", "sk-estimator"}, default="sk-label"
+        The CSS class for the inner container.
+    checked : bool, default=False
+        Whether the dropdown is checked or not. With a single estimator, we intend to
+        unfold the content.
+    doc_link : str, default=""
+        The link to the documentation for the estimator. If an empty string, no link is
+        added to the diagram. This can be generated for an estimator if it uses the
+        `_HTMLDocumentationLinkMixin`.
+    is_fitted_css_class : {"", "fitted"}
+        The CSS class to indicate whether or not the estimator is fitted or not. The
+        empty string means that the estimator is not fitted and "fitted" means that the
+        estimator is fitted.
+    is_fitted_icon : str, default=""
+        The HTML representation to show the fitted information in the diagram. An empty
+        string means that no information is shown.
+    """
+    # we need to add some padding to the left of the label to be sure it is centered
+    padding_label = "&nbsp;" if is_fitted_icon else ""  # add padding for the "i" char
+
+    out.write(
+        f'<div class="{outer_class}"><div'
+        f' class="{inner_class} {is_fitted_css_class} sk-toggleable">'
+    )
     name = html.escape(name)
 
     if name_details is not None:
         name_details = html.escape(str(name_details))
-        label_class = "sk-toggleable__label sk-toggleable__label-arrow"
+        label_class = (
+            f"sk-toggleable__label {is_fitted_css_class} sk-toggleable__label-arrow"
+        )
 
         checked_str = "checked" if checked else ""
         est_id = _ESTIMATOR_ID_COUNTER.get_id()
-        out.write(
-            '<input class="sk-toggleable__control sk-hidden--visually" '
-            f'id="{est_id}" type="checkbox" {checked_str}>'
-            f'<label for="{est_id}" class="{label_class}">{name}</label>'
-            f'<div class="sk-toggleable__content"><pre>{name_details}'
-            "</pre></div>"
+
+        if doc_link:
+            doc_label = "<span>Online documentation</span>"
+            if name is not None:
+                doc_label = f"<span>Documentation for {name}</span>"
+            doc_link = (
+                f'<a class="sk-estimator-doc-link {is_fitted_css_class}"'
+                f' target="_blank" href="{doc_link}">?{doc_label}</a>'
```

Comment:
Should we additionally include `rel="noreferrer"`? It seems like where people come from to the documentation is information that shouldn't leak. It could contain information about the hostname of the Jupyter notebook server they use, etc.